lib/services/api/error_serializer.rb

```diff
@@ -7,14 +7,15 @@ def initialize(kind, error)
       @error = error
     end
 
-    def serialize
+    def serialize(remove_sql_select = false)
       result = {
         :error => {
           :kind    => kind,
           :message => error.message,
           :klass   => error.class.name
         }
       }
+      result[:error][:message] = error.message.split(/ SELECT /i)[0] if remove_sql_select
```

Comment:
I think this regexp may need to be updated to include `UPDATE`, `INSERT` and `DELETE`.

That said, I wonder if theres a cleaner way detect SQL in the error message. Perhaps -
`error.kind_of?(PG::Error)`

Reply:
@gtanzillo Maybe I'm missing something because I do not understand the benefit of detecting if the error is a SQL error? We do not want to omit the fact that there was a PG ERROR in the API return. The part we don't want to return is the SQL statement and it's arguments.

Although I do understand  confirming that the error as a  PG error before attempting to remove the SQL statement, with its associated arguments, would be valuable.

That is to say

**I think a command of the form:** 
```bash
% curl -X GET 'https://miq-host/api/auth?requester_type=ui' -H 'Authorization: Bache' -H 'Postman-Token: <token>' -k -u "admin:<pw>"
```

**Should return:**
```bash
{"error":{"kind":"unauthorized","message":"PG::CharacterNotInRepertoire: ERROR:  invalid byte sequence for encoding \"UTF8\": 0xb5\n:","klass":"Api::AuthenticationError"}
```

**but should not return an error message like this example containing the SQL query:**
```bash
{"error":{"kind":"unauthorized","message":"PG::CharacterNotInRepertoire: ERROR:  invalid byte sequence for encoding \"UTF8\": 0xb5\n: SELECT  \"users\".* FROM \"someplace\" WHERE ...","klass":"Api::AuthenticationError"}}
```
Please let me know if I am missing something.

Thank you! JoeV